charts/job/templates/job.yaml

```diff
@@ -215,9 +188,119 @@ spec:
             {{- end }}
             {{- else }}
             {{- end }}
+        {{/* end job container */}}
+
+        {{- if .Values.cloudsqlProxy.enabled }}
```

Comment:
if someone just upgrades the chart without modifying `values.yaml` they might end with 2 sidecars, do we want put some validation and make an exclusion between `.Values.cloudsqlProxy` and .`Values.sideCar`?
or find a way to not break the release?

Reply:
The proxy is disabled by default, and the current sidecar setup is generic. It's almost always used for the sql proxy (with all the bad boilerplate), but it doesn't need to be.

Contrast to the `cloudSqlProxy` in deployments you adjusted, where the values were already explicitly for the DB connection, and people could naively upgrade the image (or be relying on a previous default) without upgrading the command.

If someone is opting into `cloudsqlProxy.enabled` and leave a fat chunk of `sideCar` configuration sitting alongside of it for that same proxy, I'd hope they notice (and they should notice when their cron doesn't run otherwise).  Same risks as someone fatfingering the script name IMO.